### PR TITLE
[MXNET-1335][fit api]Text Sentiment Classification examples using Gluon fit() API

### PR DESCRIPTION
**Pull request by karan6181**
## Description ##
This PR adds Text Sentiment Classification examples using Gluon fit() API. This PR depends on its parent PR([14346](https://github.com/apache/incubator-mxnet/pull/14346)).

JIRA epic can be found here: <https://issues.apache.org/jira/projects/MXNET/issues/MXNET-1335>

The examples has been modified from D2L Gluon [book](http://d2l.ai/chapter_natural-language-processing/index.html). This examples shows that the fit() API helps in reducing the training script by approx. 25 lines, and also the target users doesn't need to write their own training loop.

Note: We have covered some of the models, more to be added soon.  

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-1335](https://issues.apache.org/jira/projects/MXNET/issues/MXNET-1335)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by vandanavk** [comment]
@mxnet-label-bot add [Gluon, pr-work-in-progress]

**Comment by nswamy** [comment]
I think it might be a good idea to have these example(for Gluon) in the main repo, while i understand(and agree for Image classification) we don't want to replicate code from one repo to another, it could useful for people who come to mxnet directly.

Alternatively, these can be moved under integration tests.

@karan6181 Also, can you try reusing gluon-nlp constructs

**Comment by roywei** [comment]
Hi @szha, thanks for pointing out, we do plan to do a tutorial for the API. after some discussions, we dediced to move these examples to integration tests first. We will import what's available in vision model zoo, but will create these models again for RNN use cases.
Please see out updated test plan [here](https://cwiki.apache.org/confluence/display/MXNET/Gluon+Fit+API+-+Tech+Design#GluonFitAPI-TechDesign-TestPlan ). two tests for training accuracy and two teststo run all modesl (one forward and backward)

**Comment by karan6181** [comment]
Hi @szha , thank you for your inputs. we were planning to shift these examples into an integration test with some modifications(less redundant code). Would it be a good idea to use gluon-NLP toolkit API here? or constructing/preprocessing a dataset from scratch seems more appropriate?

**Comment by szha** [comment]
@karan6181 for examples you can use any dependency, license permitting.

**Comment by nswamy** [comment]
closing in lieu of https://github.com/apache/incubator-mxnet/pull/14547
